code/game/turfs/simulated/outdoors/survival_action_vr.dm

```diff
@@ -0,0 +1,113 @@
+/turf/simulated/floor/outdoors/newdirt/attack_hand(mob/user)
+	if(user.pulling)
+		return ..()
+	var/static/list/has_rocks = list("dirt5", "dirt6", "dirt7", "dirt8", "dirt9")
+	if(Adjacent(user))
+		if(icon_state in has_rocks)
+			user.visible_message("[user] loosens rocks from [src]...", runemessage = "loosens rocks from [src]")
+			if(do_after(user, 5 SECONDS, exclusive = TASK_USER_EXCLUSIVE))
+				var/obj/item/stack/material/flint/R = new(get_turf(src))
+				R.amount = rand(1,4)
+				R.pixel_x = rand(-6,6)
+				R.pixel_y = rand(-6,6)
+				icon_state = "dirt0"
+			return
+		if(locate(/obj) in src)
+			to_chat(user, "<span class='notice'>The [name] isn't clear.</span>")
+			return
+		else
+			var/choice= tgui_alert(user, "Do you want to build a growplot out of the dirt?", "Build growplot?" , list("Yes", "No"))
+			if(!choice||choice=="No")
+				return
+			user.visible_message("[user] starts piling up [src]...", runemessage = "piles up [src]")
+			if(do_after(user, 5 SECONDS, exclusive = TASK_USER_EXCLUSIVE))
+				new /obj/machinery/portable_atmospherics/hydroponics/soil(src)
```

Comment:
Optional: `if(!Adjacent(user)) return ..()` instead so it exits early rather than having it all indented